projects/kit/utils/mask/create-auto-corrected-time-pipe.ts

```diff
@@ -1,23 +1,28 @@
 import {TuiTimeMode} from '@taiga-ui/cdk';
 import {TuiTextMaskPipeHandler} from '@taiga-ui/core';
+import {MAX_TIME_VALUES} from '@taiga-ui/kit/constants';
+import {TuiTimeFormatParts} from '@taiga-ui/kit/types';
 
 /**
  * Adjusts the entered time by omitting only suitable values for hours and minutes
  * @returns time as a string
  */
 export function tuiCreateAutoCorrectedTimePipe(
     timeMode: TuiTimeMode = 'HH:MM',
+    maxValues: Record<TuiTimeFormatParts, number> = MAX_TIME_VALUES,
 ): TuiTextMaskPipeHandler {
-    const timeFormatArray: ['HH', 'MM', 'SS', 'MS'] = ['HH', 'MM', 'SS', 'MS'];
-    const maxValue = {HH: 23, MM: 59, SS: 59, MS: 999};
+    const timeFormatArray: TuiTimeFormatParts[] = ['HH', 'MM', 'SS', 'MS'];
 
     return (conformedValue: string) => {
         const indexesOfPipedChars: number[] = [];
         const conformedValueArr = conformedValue.split('');
 
         timeFormatArray.forEach(format => {
             const position = timeMode.indexOf(format);
-            const maxFirstDigit = parseInt(maxValue[format].toString().substr(0, 1), 10);
+            const maxFirstDigit =
+                format in maxValues
```

Comment:
How can `format` not be in `maxValues` if `Record` enforces all formats to be present?

Reply:
Yesterday @splincode specifically asked me to perform this check in this code and this wouldn't be necessary if such kind of enforcement worked with tokens. Right now, one can provide `maxValues: {}` through DI.

Reply:
Yes, we need to exclude non-obvious behavior at runtime with null pointer

Reply:
I'd leave it. We use Typescript and enforce strict types everywhere. If somebody provides wrong type for token that's their fault. That was the philosophy behind Taiga UI from the beginning. It's unfortunate that tokens are not type safe, that's true. But that's true across ALL tokens.

Reply:
We can use `Partial` type for token and add this to make us safer:
```ts
const max = {...MAX_TIME_VALUES, ...maxValues};
```

Reply:
Nice idea, I'll work on this today